File: app/java/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/moviescreen/impl/MovieDetailsScreenFragment.java
package com.example.moviesapp.view.screens.moviescreen.impl;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.example.moviesapp.databinding.FragmentMovieBinding;
import com.example.moviesapp.view.BaseFragment;
import com.example.moviesapp.view.ScreenView;
import com.example.moviesapp.view.screens.moviescreen.model.MovieScreenDetails;

public class MovieDetailsScreenFragment extends BaseFragment<FragmentMovieBinding> {

    @Override
    protected FragmentMovieBinding binding(LayoutInflater inflater, ViewGroup container) {
        return FragmentMovieBinding.inflate(
                inflater,
                container,
                false
        );
    }

    @Override
    protected ScreenView<FragmentMovieBinding> view() {
        return new MovieDetailsView(new MovieScreenDetails(getArguments().getString("id")));
    }
}

File: app/android/kotlin/MoviesApp/app/src/main/java/com/example/moviesapp/view/screens/moviescreen/impl/MovieDetailsScreenFragment.kt
package com.example.moviesapp.view.screens.moviescreen.impl

import android.view.LayoutInflater
import android.view.ViewGroup
import com.example.moviesapp.databinding.FragmentMovieBinding
import com.example.moviesapp.view.BaseFragment
import com.example.moviesapp.view.screens.moviescreen.model.MovieScreenDetails

internal class MovieDetailsScreenFragment: BaseFragment<FragmentMovieBinding>() {

    override fun binding(inflater: LayoutInflater, container: ViewGroup) =
            FragmentMovieBinding.inflate(
                    inflater,
                    container,
                    false
            )

    override fun view() = MovieDetailsView(
            movieScreenDetails = MovieScreenDetails(
                    arguments!!.getString("id")!!
            )
    )
}